rclcpp_components/test/test_component_manager_api.cpp

```diff
@@ -117,10 +117,10 @@ TEST_F(TestComponentManager, load_components)
       return std::find(node_names.begin(), node_names.end(), name) != node_names.end();
     };
 
-  EXPECT_TRUE(find_in_nodes("test_component_foo"));
-  EXPECT_TRUE(find_in_nodes("test_component_bar"));
-  EXPECT_TRUE(find_in_nodes("test_component_baz"));
-  EXPECT_TRUE(find_in_nodes("test_component_bing"));
+  EXPECT_TRUE(find_in_nodes("//test_component_foo"));
```

Comment:
Where does the double slash come from? A node with name `test_component_foo` and namespace `/` should have a fully qualified name `/test_component_foo`.

From http://design.ros2.org/articles/topic_and_service_names.html a topic name
```md
* must not contain any number of repeated forward slashes (`/`)
```

Reply:
When the test is being implemented, the code creates a node with a namespace of "/", rather than assigning an empty namespace. This was code written before I got involved, so I'm guessing it was the author's way to get a fully-qualified name of "/test_component_foo".

However, since the new function automatically adds in the slash, this will give us a double slash.

I could fix this by either modifying the tests, or by introducing a check to see if either namespace ends in a slash, or name begins with one, and omitting the addition of a slash in that instance. Any preference for either?

Reply:
> or by introducing a check to see if either namespace ends in a slash, or name begins with one, and omitting the addition of a slash in that instance. 

I'd prefer this one. This is what `rcl_node_t` does when it computes it's fully qualified name

https://github.com/ros2/rcl/blob/273797e387f083929a5d1abeb6b3d509ffe62f99/rcl/src/rcl/node.c#L244-L248

Reply:
Okay, got it modified. It's not my prettiest work, but it'll probably do.

I noticed a possible logic bug in the example you gave me: I think if someone enters a namespace that ends with "//", the qualified name computer will generate a qualified name with "//". But this is sort of a garbage-in-garbage-out thing, and the namespace will hopefully have been checked earlier.